apps/omg_watcher_info/lib/omg_watcher_info/supervisor.ex

```diff
@@ -63,26 +63,8 @@ defmodule OMG.WatcherInfo.Supervisor do
       ] ++ @children_run_after_repo
 
     children = [
-      {OMG.WatcherInfo.BlockApplicationConsumer, []},
       {OMG.WatcherInfo.PendingBlockProcessor, [processing_interval: Configuration.pending_block_processing_interval()]},
-      {OMG.WatcherInfo.PendingBlockQueueLengthChecker, [check_interval: Configuration.block_queue_check_interval()]},
-      {OMG.WatcherInfo.DepositConsumer, []},
-      Supervisor.child_spec(
-        {OMG.WatcherInfo.ExitConsumer, [topic: {:root_chain, "ExitStarted"}, event_type: :standard_exit]},
-        id: :std_exit_consumer
-      ),
-      Supervisor.child_spec(
-        {OMG.WatcherInfo.ExitConsumer, [topic: {:watcher, "InFlightExitStarted"}, event_type: :in_flight_exit]},
-        id: :ife_exit_consumer
-      ),
-      Supervisor.child_spec(
-        {OMG.WatcherInfo.ExitConsumer, [topic: {:watcher, "InFlightTxOutputPiggybacked"}, event_type: :in_flight_exit]},
-        id: :ife_exit_output_piggybacked_consumer
-      ),
-      Supervisor.child_spec(
-        {OMG.WatcherInfo.ExitConsumer, [topic: {:watcher, "InFlightExitOutputWithdrawn"}, event_type: :in_flight_exit]},
-        id: :ife_exit_processed_consumer
-      )
+      {OMG.WatcherInfo.PendingBlockQueueLengthChecker, [check_interval: Configuration.block_queue_check_interval()]}
```

Comment:
is there a need for this anymore? I don't think there is right? you're blocking block getter until you process the block?

Reply:
It _is_ needed. I haven't changed block consumption process, as it might fail for many reason so I want this goes with pending block. But it indeed blocks until block will be fully applied. (Besides it's easier/quicker to ensure idempotency of writes with pending block)

Reply:
the problem is that block consumption process, as it's currently, is async. the first insert is in a queue. how are you then blocking block getter to continue?